### PR TITLE
jsonschema

### PR DESCRIPTION
**Pull request by mikoxyz**

### CODE REVIEW

File: contrib/python-hatch-fancy-pypi-readme/template.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+build_style = "python_pep517"
+hostmakedepends = [
+    "python",
+    "python-build",
+    "python-hatchling",
+    "python-installer",
+    "python-wheel",
+]
+depends = [
+    "python",
```
Comment (nekopsykose) [inline contrib/python-hatch-fancy-pypi-readme/template.py:13]: don't need python also when another python dep is there, same for hostmakedepends

Comment (mikoxyz) [reply contrib/python-hatch-fancy-pypi-readme/template.py:13]: mhm, oki

File: contrib/python-hatch-fancy-pypi-readme/template.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+pkgname = "python-hatch-fancy-pypi-readme"
```
Comment (nekopsykose) [inline contrib/python-hatch-fancy-pypi-readme/template.py:1]: we patched this out everywhere so far, but i guess we could just keep it, since it's a tiny module and saves some patches (with no downside to having it in build hostdeps)

File: contrib/python-jsonschema/template.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python",
+    "python-build",
+    "python-hatch-fancy-pypi-readme",
+    "python-hatch_vcs",
+    "python-hatchling",
+    "python-installer",
+    "python-pytest",
```
Comment (nekopsykose) [inline contrib/python-jsonschema/template.py:12]: pytest in checkdepends for all of them

File: contrib/python-maturin/template.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+def post_patch(self):
+    from cbuild.util import cargo
+
+    self.cargo = cargo.Cargo(self)
+    self.cargo.vendor()
+    cargo.setup_vendor(self)
+
+
+def post_install(self):
+    self.install_license("license-apache")
```
Comment (nekopsykose) [inline contrib/python-maturin/template.py:38]: don't need to install apache

File: contrib/python-referencing/template.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+pkgver = "0.32.1"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python",
+    "python-build",
+    "python-hatch_vcs",
+    "python-hatchling",
+    "python-installer",
+    "python-rpds",
```
Comment (nekopsykose) [inline contrib/python-referencing/template.py:11]: this doesn't need to be in hostdeps, you can pass --skip-dependency-check in make_build_args if it's build that complains

File: contrib/python-referencing/template.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "python-attrs",
+    "python-rpds",
+]
+pkgdesc = "Cross-specification JSON referencing"
+maintainer = "miko <mikoxyzzz@gmail.com>"
+license = "MIT"
+url = "https://github.com/python-jsonschema/referencing"
+source = f"{url}/releases/download/v{pkgver}/referencing-{pkgver}.tar.gz"
+sha256 = "3c57da0513e9563eb7e203ebe9bb3a1b509b042016433bd1e45a2853466c3dd3"
+# test_sanity.py causes a circular dependency since it depends on jsonschema, which depends on this package
```
Comment (nekopsykose) [inline contrib/python-referencing/template.py:25]: you could --deselect this one test in that case

File: contrib/python-hatch-fancy-pypi-readme/template.py (hunk 2 above)
Comment (q66) [reply contrib/python-hatch-fancy-pypi-readme/template.py:1]: i'd prefer patching it out, since it's entirely useless, and it's very little work to patch it out

File: contrib/python-jsonschema-specifications/template.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+# get rid of this once pep517's check stuff is fixed up
+def pre_check(self):
+    whl = list(
+        map(
+            lambda p: str(p.relative_to(self.cwd)),
+            self.cwd.glob(self.make_install_target),
+        )
+    )
+
+    self.do(
```
Comment (q66) [inline contrib/python-jsonschema-specifications/template.py:33]: do we really need this venv nonsense for every single one of these? because it hasn't been needed for literally anything until now

File: contrib/python-maturin/template.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+pkgname = "python-maturin"
```
Comment (q66) [inline contrib/python-maturin/template.py:1]: on a second thought, split specifically this into its own PR and leave the others in this one

Comment (q66) [reply contrib/python-maturin/template.py:1]: while at it, this does not bother to properly set features (`--features=full,native-tls,password-storage`), which need to be set (because we don't want to build with rustls)

File: contrib/python-jsonschema-specifications/template.py (hunk 7 above)
Comment (nekopsykose) [reply contrib/python-jsonschema-specifications/template.py:33]: no, just the native-module ones. but you could make it the default for do_check and it would be an improvement like i explained yesterday

Comment (nekopsykose) [reply contrib/python-jsonschema-specifications/template.py:33]: > no, just the native-module ones.  

(or to be specific- it depends on the build-backend, setuptools keeps it in build/lib.* too. but maturin does not persist a whole pythonpath import on disk and only gives you a wheel. with a venv though you can always 'just install the wheel' and not care about any of the specifics, like how in some templates we add PYTHONPATH to the build/lib folder, etc)

**Comment by q66** [comment]
i've merged maturin and i've done the venv stuff for check earlier too, so you can remove all that and rebase this

File: contrib/python-jsonschema-specifications/template.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    "python-hatch_vcs",
+    "python-hatchling",
+    "python-installer",
+]
+checkdepends = [
+    "python-referencing",
+    "python-pytest",
+]
+depends = ["python-referencing"]
+pkgdesc = "Json Schema meta-schemas and vocabularioes, exposed as a Registry"
```
Comment (q66) [inline contrib/python-jsonschema-specifications/template.py:16]: vocabularioes

File: contrib/python-rpds/template.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+url = "https://github.com/crate-py/rpds"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "10d04e7160e5973c97bc371c1bb26b10d91385d532d4b9d13d724ff4d4f23819"
+
+
+def post_patch(self):
+    from cbuild.util import cargo
+
+    self.cargo = cargo.Cargo(self)
+    self.cargo.vendor()
```
Comment (q66) [inline contrib/python-rpds/template.py:33]: cargo.Cargo(self).vendor()

**Comment by q66** [comment]
also prefer $(PYPI_SITE) for source urls, like most of the other python packages
